### PR TITLE
search domain discovery: add whois lookup fallback

### PR DESCRIPTION
**Pull request by FR4NK-W**
Add fallback feature for discovering a valid search domain by looking up the contact information for the external public IP and deriving potential search domains from the contact addresses.

### CODE REVIEW

File: hinting/dns_search_domain_fallbacks_whois.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	rirWHOIS  = []string{
+		"whois.afrinic.net",
+		"whois.lacnic.net",
+		"whois.apnic.net",
+		"whois.ripe.net",
+		"whois.arin.net",
+	}
+)
+
+func reverseLookupWhois(addr netip.Addr) (domains []string) {
```
Comment (marcfrei) [inline hinting/dns_search_domain_fallbacks_whois.go:23]: `Whois` > `WHOIS` for consistency with the names of the variables above?

File: hinting/dns_search_domain_fallbacks_whois.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	response, err := resolveWhoisRedirects(addr, ianaWHOIS)
+	if err != nil {
+		return
+	}
+	return extractEmailDomains(response)
+}
+
+func resolveWhoisRedirects(addr netip.Addr, server string) (response string, err error) {
```
Comment (marcfrei) [inline hinting/dns_search_domain_fallbacks_whois.go:31]: `Whois` > `WHOIS`?

**Review by marcfrei — APPROVED** [review]
LGTM

File: hinting/dns_search_domain_fallbacks_whois.go (hunk 1 above)
Comment (FR4NK-W) [reply hinting/dns_search_domain_fallbacks_whois.go:23]: Done.

File: hinting/dns_search_domain_fallbacks_whois.go (hunk 2 above)
Comment (FR4NK-W) [reply hinting/dns_search_domain_fallbacks_whois.go:31]: Done.

File: hinting/dns_search_domain_fallbacks_whois.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+				whoisServer = whoisRefer
+				continue
+			}
+		}
+		break
+	}
+	return
+}
+
+func queryWhois(serverTCPAddr *net.TCPAddr, queryAddr netip.Addr) (response string, err error) {
```
[the remark this answers is not in the document]
Comment (FR4NK-W) [reply hinting/dns_search_domain_fallbacks_whois.go:72]: Done.

**Comment by FR4NK-W** [comment]
rebased

**Pull request merged**